### PR TITLE
Detect your own endpoint in HeightsByEndpoints()

### PR DESCRIPTION
**Pull request by yacovm**
Currently, when an orderer node chooses a single node endpoint to pull blocks from (and the rest to pull headers from), it may pick itself, which is a problem because this endpoint will never bring it any blocks since it is itself.

This commit makes the function HeightsByEndpoint return the endpoint of yourself, such that the caller can filter it out.

The identification of your own endpoint is done by looking at the TLS certificate of the remote node and comparing it with your own.

### CODE REVIEW

File: orderer/common/cluster/deliver.go [hunk 1]
```diff
@@ -115,15 +116,26 @@ func (p *BlockPuller) PullBlock(seq uint64) *common.Block {
...
 
 // HeightsByEndpoints returns the block heights by endpoints of orderers
-func (p *BlockPuller) HeightsByEndpoints() (map[string]uint64, error) {
+func (p *BlockPuller) HeightsByEndpoints() (map[string]uint64, string, error) {
 	endpointsInfo := p.probeEndpoints(0)
 	res := make(map[string]uint64)
+	var myEndpoint string
+
+	var myCert *x509.Certificate
+	if len(p.TLSCert) > 0 {
```
Comment (tock-ibm) [inline orderer/common/cluster/deliver.go]: This cannot fail? no need to handle the error?

File: orderer/common/cluster/deliver.go [hunk 2]
```diff
@@ -115,15 +116,26 @@ func (p *BlockPuller) PullBlock(seq uint64) *common.Block {
...
+		myCert, _ = x509.ParseCertificate(p.TLSCert)
+	}
+
 	for endpoint, endpointInfo := range endpointsInfo.byEndpoints() {
+		if myCert != nil && reflect.DeepEqual(myCert, endpointInfo.certificate) {
```
Comment (tock-ibm) [inline orderer/common/cluster/deliver.go]: Just a nit: endpointInfo shadows the name of the struct

```
Declaration of 'endpointInfo' shadows declaration at deliver.go 
```
L531

File: orderer/common/cluster/deliver.go [hunk 3]
```diff
@@ -115,15 +116,26 @@ func (p *BlockPuller) PullBlock(seq uint64) *common.Block {
...
+			myEndpoint = endpoint
+		}
 		endpointInfo.conn.Close()
 		res[endpoint] = endpointInfo.lastBlockSeq + 1
 	}
+
 	p.Logger.Info("Returning the heights of OSNs mapped by endpoints", res)
-	return res, endpointsInfo.err
+	return res, myEndpoint, endpointsInfo.err
```
Comment (tock-ibm) [inline orderer/common/cluster/deliver.go:134]: the log above should be `Infof()`,  and add `myEndpoint` as well

**Review by tock-ibm — COMMENTED** [review]
Looks good, just a few nits

File: orderer/common/cluster/deliver.go (hunk 1 above)
Comment (yacovm) [reply orderer/common/cluster/deliver.go:125]: why handle the error? if our TLS cert cannot be parsed we have bigger problems, no?

File: orderer/common/cluster/deliver.go (hunk 2 above)
Comment (yacovm) [reply orderer/common/cluster/deliver.go]: i know, but this is irrelevant to this PR.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**